python/feathub/processors/flink/table_builder/flink_table_builder.py

```diff
@@ -234,16 +234,26 @@ def _get_table_from_derived_feature_view(
             Tuple[str, Sequence[str]], Dict[str, JoinFieldDescriptor]
         ] = {}
 
+        # This list contains all ExpressionTransform features listed after the first
+        # OverWindowTransform or JoinTransform feature in the dependent_features.
+        # These features are evaluated after all over windows and before table join.
+        expression_features_following_first_over_window_or_join = []
+
         for feature in dependent_features:
             if feature.name in tmp_table.get_schema().get_field_names():
                 continue
             if isinstance(feature.transform, ExpressionTransform):
-                tmp_table = self._evaluate_expression_transform(
-                    tmp_table,
-                    feature.transform,
-                    feature.name,
-                    feature.dtype,
-                )
+                if len(right_tables) > 0 or len(window_agg_map) > 0:
+                    expression_features_following_first_over_window_or_join.append(
```

Comment:
Should we also evaluate JoinTransform features before OverWindowTransform features? Can we also add a unit test that shows OverWindowTransform features can depend on JoinTransform features?

Reply:
This PR also made the change to value JoinTransform before OverWindowTransform. I renamed the unit test so that it is clear that it tests OverWindowTransform with JoinTransform features.